components/patient-measurements/migrations/src/main/java/org/phenotips/measurements/internal/R71291PhenoTips1969DataMigration.java

```diff
@@ -0,0 +1,270 @@
+
+    /** Resolves class names to the current wiki. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<EntityReference> entityResolver;
+
+    @Override
+    public String getDescription()
+    {
+        return "Migrate existing measurements to the new format";
+    }
+
+    @Override
+    public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(71291);
+    }
+
+    @Override
+    public void hibernateMigrate() throws DataMigrationException, XWikiException
+    {
+        getStore().executeWrite(getXWikiContext(), this);
+    }
+
+    @Override
+    public Object doInHibernate(Session session) throws HibernateException, XWikiException
+    {
+        XWikiContext context = getXWikiContext();
+        XWiki xwiki = context.getWiki();
+        DocumentReference patientClassReference =
+            R71291PhenoTips1969DataMigration.this.entityResolver.resolve(PATIENT_CLASS);
+        DocumentReference oldClassReference =
+            R71291PhenoTips1969DataMigration.this.entityResolver.resolve(MEASUREMENTS_CLASS);
+        DocumentReference newClassReference =
+            R71291PhenoTips1969DataMigration.this.entityResolver.resolve(MEASUREMENT_CLASS);
+
+        Query q =
+            session.createQuery("select distinct o.name from BaseObject o where o.className = '"
+                + R71291PhenoTips1969DataMigration.this.serializer.serialize(oldClassReference) + "'");
+
+        @SuppressWarnings("unchecked")
+        List<String> documents = q.list();
+
+        for (String docName : documents) {
+            XWikiDocument doc =
+                xwiki.getDocument(R71291PhenoTips1969DataMigration.this.resolver.resolve(docName), context);
+            List<BaseObject> oldObjects = doc.getXObjects(oldClassReference);
+            if (oldObjects.isEmpty()) {
+                continue;
+            }
+
+            BaseObject patientObject = doc.getXObject(patientClassReference);
+            Date birth = null;
+            if (patientObject != null) {
+                birth = patientObject.getDateValue("date_of_birth");
+            }
+
+            for (BaseObject object : oldObjects) {
+                Date date = object.getDateValue(DATE_FIELD);
+                String age = calculateAge(birth, date);
+                Collection fieldList = object.getFieldList();
+                migrateObject(date, age, fieldList, doc, context, newClassReference);
+                doc.removeXObject(object);
+            }
+            doc.setComment("Migrated MeasurementsClass data into MeasurementClass instances");
+            doc.setMinorEdit(true);
+            try {
+                // There's a bug in XWiki which prevents saving an object in the same session that it was loaded,
+                // so we must clear the session cache first.
+                session.clear();
+                ((XWikiHibernateStore) getStore()).saveXWikiDoc(doc, context, false);
```

Comment:
Just curious do we always have to cast? Seems like saveXWikiDoc() is the one called every time, why dont we return appropriate store from getStore() in the first place? (not commenting on this commit, just a general question)?